File: 1. Building Abstraction with Procedures/1.1 The Elements of Programming/square/square.go
// SICP 1.1.4
// Compound Procedures

package main

import "fmt"

func square(x int) int {
	return x * x
}

func sum_of_squares(x, y int) int {
	return square(x) + square(y)
}

func main() {
	fmt.Println(square(4))
	fmt.Println(sum_of_squares(3, 4))
}

File: 1. Building Abstraction with Procedures/1.1 The Elements of Programming/square/square.swift
// SICP 1.1.4
// Compound Procedures

func square(_ x: Int) -> Int {
    x * x
}

func sum_of_squares(_ x: Int, _ y: Int) -> Int {
    square(x) + square(y)
}

print(square(4))
print(sum_of_squares(3, 4))
